docusaurus/docs/iOS/guides/working-with-attachments.md

```diff
@@ -1,3 +1,263 @@
+    }
+
+    override func contentViewDidUpdateContent() {
+        attachmentView.content = attachments(payloadType: FileAttachmentPayload.self).first
+    }
+}
+
+
+class MyCustomAttachmentView: _View {
+    var content: ChatMessageFileAttachment? {
+        didSet { updateContentIfNeeded() }
+    }
+
+    override func setUpAppearance() {
+        super.setUpAppearance()
+    }
+
+    override func setUpLayout() {
+        super.setUpLayout()
+    }
+
+    override func updateContent() {
+        super.updateContent()
+    }
+}
+
+```
+
+### Register the custom class to be used for file attachments
+
+The last step is to inform the SDK to use `MyCustomAttachmentViewInjector` for file attachments instead of the default one. This is something that you want to do as early as possible in your application life-cycle.
+
+```swift
+Components.default.filesAttachmentInjector = MyCustomAttachmentViewInjector.self
+```
+
+## How to build a custom attachment
+
+Stream chat allows you to create your own types of attachments as well. The steps to follow to add support for a custom attachment type are the following:
+
+1. Extend `AttachmentType` to include the custom type
+1. Create a new AttachmentPayload struct to handle your custom attachment data
+1. Create a new AttachmentViewInjector 
+1. Configure the SDK to use your view injector class to render custom attachments
+
+Let's assume we want to attach a workout session to a message, the payload of the attachment will looks like this:
+
+
+```json
+{
+  "type": "workout",
+  "image_url": "https://path.to/some/great/picture.png",
+  "workout-type": "walk",
+  "workout-energy-cal": 75500,
+  "workout-distance-meters": 1412,
+  "workout-duration-seconds": 840
+}
+```
+
+Here's how we get around the first two steps:
+
+```swift
+public extension AttachmentType {
+    static let workout = Self(rawValue: "workout")
+}
+
+public struct WorkoutAttachmentPayload: AttachmentPayload {
+    public static var type: AttachmentType = .workout
+
+    var imageURL: URL
+    var workoutDistanceMeters: Int?
+    var workoutType: String?
+    var workoutDurationSeconds: Int?
+    var workoutEnergyCal: Int?
+
+    private enum CodingKeys: String, CodingKey {
+        case workoutDistanceMeters = "workout-distance-meters"
+        case workoutType = "workout-type"
+        case workoutDurationSeconds = "workout-duration-seconds"
+        case workoutEnergyCal = "workout-energy-cal"
+        case imageURL = "image_url"
+    }
+}
+```
+
+Here we extended `AttachmentType` to include the `workout` type and afterwards we introduced a new struct to match the payload data that we expect. 
+
+1. WorkoutAttachmentPayload.type is used to match message attachments to this struct
+1. All attachment fields are optional, this is highly recommended for all custom data
+1. We use CodingKeys to map JSON field names to struct fields
+
+Let's now create a custom view injector to handle the workout attachment. 
+
+```swift
```

Comment:
Typically, it is good to have comments in code blocks of this size giving a brief explanation of what each block accomplishes even if only to make the code block have portions that stand out when somebody uses cmd/ctr+f on the page.